src/react-ruler/Ruler.tsx

```diff
@@ -127,164 +141,224 @@ export default class Ruler extends React.PureComponent<RulerProps> implements Ru
     context.lineWidth = 1;
     context.font = font;
     context.fillStyle = textColor;
+    context.textBaseline = this.getCanvasTextBaseline();
+    context.translate(0.5, 0);
+    context.beginPath();
+  }
 
-    switch (direction) {
+  private getCanvasTextBaseline(): CanvasTextBaseline {
+    switch (this.props.direction) {
       case 'start':
-        context.textBaseline = 'top';
-        break;
+        return 'top';
       case 'center':
-        context.textBaseline = 'middle';
-        break;
+        return 'middle';
       case 'end':
-        context.textBaseline = 'bottom';
-        break;
+        return 'bottom';
     }
+  }
 
-    context.translate(0.5, 0);
-    context.beginPath();
-
-    const size = isHorizontal ? width : height;
-    const zoomUnit = nextZoom * unit;
-    const minRange = Math.floor((scrollPos * nextZoom) / zoomUnit);
-    const maxRange = Math.ceil((scrollPos * nextZoom + size) / zoomUnit);
-    const length = maxRange - minRange;
-    const alignOffset = Math.max(['left', 'center', 'right'].indexOf(textAlign) - 1, -1);
-    const barSize = isHorizontal ? height : width;
+  private renderRangeBackground(renderOptions: RulerRenderOptions) {
+    const { isHorizontal, containerSize, scrollPos, zoom } = renderOptions;
+    const { range = [-Infinity, Infinity], rangeBackgroundColor } = this.props;
+    const context = this._canvasContext;
 
-    // Draw Range Background
     if (rangeBackgroundColor !== 'transparent' && range[0] !== -Infinity && range[1] !== Infinity) {
```

Comment:
I think you can move this condition to the method, in such case it will improve readability and will simplify the current function